### PR TITLE
Wgsl switch requires default

### LINKED ISSUE #714: Is a switch default clause always required?
This was raised by https://github.com/gpuweb/gpuweb/pull/713

We need to say what happens when a switch selector expression evaluates to something other than any of the case literals.

Since we have a desire to avoid undefined behaviour, I **propose that a switch statement must always have a default clause**.

The alternative is to make a rule that control skips past the switch statement entirely, but that's quite non-traditional and hence inadvisable.

### PR DESCRIPTION
**Pull request by dneto0**
Fixes #714

Builds on #713

### CODE REVIEW

**Comment by dneto0** [comment]
Queueing this up for review after #713

File: wgsl/index.bs [hunk 1]
```diff
@@ -1039,15 +1039,44 @@ switch_stmt
...
+transferred to the `default` clause.
+
+Each switch statement must have exactly one default clause.
+
+A literal value must not appear more than once in the case selectors for a switch statement.
+
+The case selector values must have the same type as the selector expression.
+
+Note: A literal value must not be duplicated within a case clause, and must not appear in more
+than one case selector list.
```
Comment (grorg) [inline wgsl/index.bs:1069]: Isn't this exactly what you said above? "A literal value must not appear more than once in the case selectors for a switch statement.". I think that covers duplication within a single case, as well as appearing in only one case.

File: wgsl/index.bs [hunk 2]
```diff
@@ -1039,15 +1039,44 @@ switch_stmt
...
+
+Note: The value of the literal is what matters, not the spelling. For example `0` and `00` both
+denote the zero value.
+
```
Comment (grorg) [inline wgsl/index.bs:1073]: If you're going to give examples, maybe also provide a decimal v hex one where it is a more different spelling?

File: wgsl/index.bs (hunk 1 above)
Comment (dneto0) [reply wgsl/index.bs:1069]: ok. will remove. (This was already merged as part of the parent.)

File: wgsl/index.bs (hunk 2 above)
Comment (dneto0) [reply wgsl/index.bs]: Added 0x0000 as another example spelling of 0.

**Comment by dneto0** [comment]
PTAL

**Comment by dj2** [comment]
If you can fixup the merge conflicts here I can commit.

**Comment by dneto0** [comment]
Ok, this is now rebased.

I also added the rule that there must be exactly one default clause.  We had missed that before.

File: wgsl/index.bs [hunk 3]
```diff
@@ -1039,15 +1039,41 @@ switch_stmt
...
 
+A switch statement transfers control to one of a set of case clauses, or to a default clause,
+depending the evaluation of a selector expression of a scalar integer type.
+
+If the selector value equals a value in a case selector list, then control is transferred to
+the body of that case clause.
+If the selector value does not equal any of the case selector values, then control is
+transferred to the `default` clause.
+
+Each switch statement must have exactly one default clause.
```
Comment (dj2) [inline wgsl/index.bs:1062]: Is this really a `must`? Shouldn't it be `may`? The default case isn't required is it?

Comment (dneto0) [reply wgsl/index.bs:1101]: It's a must because of #714

**Comment by dneto0** [comment]
Whoops, had some branch confusion.
This is rebased again.

**Pull request merged**